cloudinit/distros/photon.py

```diff
@@ -99,26 +98,23 @@ def _bring_up_interfaces(self, device_names):
         return ret
 
     def _write_hostname(self, hostname, filename):
-        conf = None
-        try:
-            # Try to update the previous one
-            # Let's see if we can read it first.
-            conf = HostnameConf(util.load_file(filename))
-            conf.parse()
-        except IOError:
-            pass
-        if not conf:
-            conf = HostnameConf('')
-        conf.set_hostname(hostname)
-        util.write_file(filename, str(conf), mode=0o644)
+        if filename.endswith('/previous-hostname'):
+            util.write_file(filename, hostname)
+        else:
+            _ret, _out, _err = self.exec_cmd(['hostnamectl', 'set-hostname',
+                                             str(hostname)])
```

Comment:
```suggestion
            self.exec_cmd(['hostnamectl', 'set-hostname', str(hostname)])
```
If we're not using the results, then we shouldn't store them.

Reply:
Agree.